pkg/asset/agent/image/agentartifacts.go

```diff
@@ -51,32 +53,37 @@ func (a *AgentArtifacts) Generate(dependencies asset.Parents) error {
 	kargs := &Kargs{}
 	baseIso := &BaseIso{}
 	agentManifests := &manifests.AgentManifests{}
+	agentClusterInstall := &manifests.AgentClusterInstall{}
 	registriesConf := &mirror.RegistriesConf{}
 	agentconfig := &config.AgentConfig{}
 
-	dependencies.Get(ignition, kargs, baseIso, agentManifests, registriesConf, agentconfig)
+	dependencies.Get(ignition, kargs, baseIso, agentManifests, agentClusterInstall, registriesConf, agentconfig)
 
 	ignitionByte, err := json.Marshal(ignition.Config)
 	if err != nil {
 		return err
 	}
+
 	a.CPUArch = ignition.CPUArch
 	a.RendezvousIP = ignition.RendezvousIP
 	a.IgnitionByte = ignitionByte
 	a.ISOPath = baseIso.File.Filename
 	a.Kargs = kargs.KernelCmdLine()
+
 	if agentconfig.Config != nil {
 		a.BootArtifactsBaseURL = strings.Trim(agentconfig.Config.BootArtifactsBaseURL, "/")
 	}
 
-	agentTuiFiles, err := a.fetchAgentTuiFiles(agentManifests.ClusterImageSet.Spec.ReleaseImage, agentManifests.GetPullSecretData(), registriesConf.MirrorConfig)
-	if err != nil {
-		return err
-	}
+	if agentClusterInstall.GetExternalPlatformName() != string(models.PlatformTypeOci) {
+		agentTuiFiles, err := a.fetchAgentTuiFiles(agentManifests.ClusterImageSet.Spec.ReleaseImage, agentManifests.GetPullSecretData(), registriesConf.MirrorConfig)
+		if err != nil {
+			return err
+		}
 
-	err = a.prepareAgentArtifacts(a.ISOPath, agentTuiFiles)
-	if err != nil {
-		return err
+		err = a.prepareAgentArtifacts(a.ISOPath, agentTuiFiles)
```

Comment:
`prepareAgentArtifacts` must be executed always. It's ok to pass an empty slice to avoid adding agent-tui.